src/sentry/static/sentry/app/views/performance/transactionSummary/index.tsx

```diff
@@ -175,23 +175,25 @@ function generateSummaryEventView(
   // Use the user supplied query but overwrite any transaction or event type
   // conditions they applied.
   const query = decodeScalar(location.query.query) || '';
-  const conditions = Object.assign(tokenizeSearch(query), {
-    'event.type': ['transaction'],
-    transaction: [transactionName],
-  });
+  const conditions = tokenizeSearch(query);
+  conditions.setTag('event.type', ['transaction']);
+  conditions.setTag('transaction', [transactionName]);
```

Comment:
Should `setTag()` return `this` so you could do `conditions.setTag().setTag()` etc?

Reply:
I thought about that. I can change that, it would probably make things cleaner.

Reply:
All the mutation methods on QueryResults could return `this`. That would give you a more fluent interface to build/modify query conditions.